samza-core/src/main/java/org/apache/samza/container/ContainerHeartbeatClient.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.container;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Random;
+import java.util.stream.Collectors;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class ContainerHeartbeatClient {
+  private static Logger log = LoggerFactory.getLogger(ContainerHeartbeatClient.class);
+  private final String coordinatorUrl;
+  private final String executionEnvContainerId;
+  private final int numberOfRetries = 3;
+  private final int timeout = 5000;
+
+  public ContainerHeartbeatClient(String coordinatorUrl, String executionEnvContainerId) {
+    this.coordinatorUrl = coordinatorUrl;
+    this.executionEnvContainerId = executionEnvContainerId;
+  }
+
+  public boolean isAlive() {
```

Comment:
Is there any reason why you chose to return a boolean for isAlive, instead of returning the ContainerHeartbeatResponse? 

When I think about a "client" of any kind, it only handles the request/response part. Not the functional details associated with the caller. 

For example, Line 48 in ContainerHeartbeatMonitor -> `containerHeartbeatClient.isAlive()` It is not clear if it is returning the "liveness" of the client or the "liveness" of the container ? 

If you return the response object directly, it will be cleaner. It will also be more re-usable, if we chose to fetch other metadata in the same protocol.